### ISSUE TITLE
Warning when compiling

### ISSUE THREAD

**Issue by jtrees** [elridion/gelfx#2]
When compiling the dependencies for my project I get a warning.

```
==> gelfx
Compiling 2 files (.ex)
warning: JSON library Jason is not available - :nofile
  /foo/deps/gelfx/lib/gelfx.ex:1: (module)

Generated gelfx app
```

```
# mix.exs

defmodule Vortex.Mixfile do
  # ...
  defp deps do
    [
      {:gelfx, "~> 1.0"},
      {:jason, "~> 1.1"},
      # ...
    ]
  end
end
```

```
# mix.lock

%{
  "gelfx": {:hex, :gelfx, "1.1.0", "86123fbfa63a0814f38330050e25e15a2ab3b9184fcf1995526c449c5fae9573", [:mix], [], "hexpm", "6bfcaa6ffe416a1291bbe00695c908a94265b4681c90224c64657ff1ae4314c1"},
  "jason": {:hex, :jason, "1.1.2", "b03dedea67a99223a2eaf9f1264ce37154564de899fd3d8b9a21b1a6fd64afe7", [:mix], [{:decimal, "~> 1.0", [hex: :decimal, repo: "hexpm", optional: true]}], "hexpm", "fdf843bca858203ae1de16da2ee206f53416bbda5dc8c9e78f43243de4bc3afe"},
  # ...
}
```

```
$ elixir --version            
Erlang/OTP 21 [erts-10.0.5] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe]

Elixir 1.10.4 (compiled with Erlang/OTP 21)
```

**Comment by elridion** [comment]
Yea it's kind of a known issue.
It happens since gelfx tries to check if the defined `json_libary`is available during compilation.

This fails in (most) build environments that newly fetch all dependencies.
If the configured `json_libary` is in fact present but compiled afterwards it yields an warning.

To fix the the `__after_compile__` check should be moved into the `gen_stage` initialization.

**Comment by elridion** [comment]
Fixed in 80aa8ba and included in release 1.1.2

**Comment by jtrees** [comment]
Sweet! Thanks.
